### ISSUE TITLE
data disappearing on /chemical/ page

### ISSUE THREAD

**Issue by kdionisio** [HumanExposure/factotum#1260]
There is a problem with 'missing' data not appearing on the /chemical/ page.

As an example for http://factotum.epa.gov/chemical/DTXSID7020637/, there are existing products linked to PUCs which are not appearing in the summary.

Specifically the 'Personal care' PUC was appearing in both the PUC list and the PUC bubble plot (there *are* known products linked to personal care PUCs which contain formaldehyde). However, after assigning a few hundred products into the 'personal care' PUC, this PUC disappeared from both the PUC list and the PUC bubble plot.

Definition of Done:
All PUCs containing data should appear in PUC list and in bubble plot on /chemical/ page

**Comment by debboutr** [comment]
HERE are a couple of issues that I have found while investigating this issue, the first might be what leads to the bug, though I still am not sure.

one issue, which we have had to deal with before is due to having null=True on CharFields in models, django docs mention not to do this, which we have talked about before but it looks like it is biting us again. If a chemical is curated and the raw_cas value is NULL, normally if they edit the raw_cas or raw_chem_name fields we delete the ForeignKey to DSSTOXLookup and the rid field, but when editing the form on an instance like this, even if you just change another field, say component, the chemical gets uncurated because the `raw_cas` value from the form is an empty string whereas the initial value was NULL so the field tracker sees it as being changed and uncurates the chemical. 

https://docs.djangoproject.com/en/2.2/ref/models/fields/#null
-- this is why they don't recommend to have null=True

It also looks like there are chemical records that were brought in that hold a value for min, central, and max_composition and we have since added the constraint to only have min/max or just central , here is a url on prod that will illustrate, try to edit citric acid
http://127.0.0.1:8000/datadocument/1369930/#chem-1439391 
we may need to do a data migration for this.

**Comment by debboutr** [comment]
Based on what I have to look at between the 2 prod dumps

- prod_factotum_2019-11-11.sql
- prod_factotum_2019-12-16.sql

I'm not seeing a link of any "Personal Care" PUCs to Formaldehyde <DTXSID7020637> in either of the dumps, which I believe would be before and after this issue arose. I checked the AuditLog to see what that might tell me, but we don't have triggers on the ProductDocument model and the ones that we do have on the RawChem model don't cover the dsstox foreign key. 

I only have these 2 checkpoints to go off of so I could be missing something. There is one chemical that is linked to Formaldehyde in the older dump of prod that isn't linked in the new one, it is not linked to any product at the time of the old dump, but is linked to a product in the new dump.

here is the link to that one
[https://factotum.epa.gov/datadocument/273447/#chem-1389428](`https://factotum.epa.gov/datadocument/273447/#chem-1389428` )

There is the unlinking bug that I describe in the commen above which has potential to have influenced it, but w/o my ability to seeing a link in the old dump of prod I can't check into the audit log when or how the chem was unlinked if that's the reason.

I don't know how else to approach finding this at the moment, so if you have any ideas I'd be happy to hear them.

thanks.

**Comment by kdionisio** [comment]
PUtting on hold to deal with larger data issue re. nulls/blanks in data fields

**Comment by kdionisio** [comment]
check if still an issue after #1298 is complete

**Comment by kdionisio** [comment]
No longer seeing this bug after pushing Case sprint and after merging in large batch of curated chemicals.
